### DISCUSSION TITLE
No more tray icon with 4.0.1

### DISCUSSION THREAD

**Discussion by Npa972** [transmission/transmission#5065] [Q&A] [answered]
Hi i don't see any option (show transmission icon in the notification area) in the 4.0.1 version
OS : archlinux
Desktop : xfce
How it's works now ?
Thank you

**Comment by GaryElshaw** [comment]
It has to be built with GTK3 rather than 4.

**Reply by Npa972** [reply]
Archlinux package is built with gtk4.
Thank's

**Reply by JPFrancoia** [reply]
Archlinux + cinnamon here. Indeed, I can't minimize Transmission v4. I had to downgrade Transmission for now. Is there a workaround?

**Reply by GaryElshaw** [reply]
Building from source with GTK3.

**Comment by ckerr** [comment, marked as answer]
GtkStatusIcon was removed in GTK 4. If you want systray icons report take up the issue upstream; this is a desktop decision rather than a Transmission one.

These two pages [1](https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/84), [2](https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/191) discusses some of the background.

**Reply by Npa972** [reply]
OK thanks you for your answer.

**Reply by malbajun** [reply]
> GtkStatusIcon was removed in GTK 4. If you want systray icons report take up the issue upstream; this is a desktop decision rather than a Transmission one.
> 
> These two pages [1](https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/84), [2](https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/191) discusses some of the background.

But then why you dont aplly the feature for example as answered by NN (Neuronet, formerly AI):

> Background Processing: If your application needs to stay alive after the user closes the window, you should keep the process running and simply spawn new windows from GtkApplication when the user re-launches it.

**Comment by mikedld** [comment]
Transmission can also use libappindicator but GTK 4 support isn't available there yet either. See AyatanaIndicators/libayatana-appindicator#22 if you want to contribute.

**Comment by bbhtt** [comment]
Will it be possible to add support for background portal to keep the app running in background (ie. part of the tray icon functionality is to keep the app alive hidden from view) https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.Background? GNOME 44 adds a new UI for controlling background apps.

Should this be a new feature request?

**Reply by mks-h** [reply]
Second this. Having to keep the Transmission window open all the time is a nuisance. Just having it run in a background (and consequentially be visible in the "Background apps" of the upcoming GNOME 44) is really all I ask for.

**Reply by ckerr** [reply]
To say upfront -- https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/191 is really long, so I haven't read it all and I could be wrong -- but it looks like this is still in the proposal phase? Even if this functionality were requested for transmission-gtk _and_ if someone wanted to work on it, we should probably wait until the upstream proposal is finalized.

In the interim, if the goal is to have Transmission running all the time with or without a GUI, one alternative is using `transmission-daemon` + a remote GUI like `transmission-qt` until something more concrete shows up upstream.

**Reply by mks-h** [reply]
The issue you've linked is just a redesign of the existing way to represent the background portal in GNOME. 

Right now, from users' perspective, when an app's window is closed but it continues to run in the background, users will be prompted with a notification whether to allow or disallow it to continue. If the user disallows, the app will be force-quit. If the user allows or ignores the notification, the app will continue to run.

However, [an initial implementation](https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2624) of the linked issue is present in the upcoming GNOME 44 that, instead of the current behavior, it will just show a list of running background applications in the quick settings.

P.S. Thanks for replying. Running transmission-daemon is not always a solution, since installing packages is not a good way to go on immutable distributions like Fedora Silverblue, Vanilla OS, Steam OS, etc. Although it is possible on some of them, but generally such distributions prefer users to install software through the Flatpak packages.

**Comment by Philzen** [comment]
There is also a `transmission-qt` version available via pacman. I noticed the tray minimization feature was there when i started using Transmission some years ago, but after some update it was gone, now i know presumably b/c gtk changed from v3 to v4.

Now i installed `transmission-qt` and the tray minimization feature is back again in my XFCE tray :rocket:

**Comment by isness** [comment]
I just use 3.0.0 version on Gnome 45. There is no point in updating to the latest version when one of the main  features is missing, no matter the cause. I just need to it work and be usable, and, honestly, for me and a lot of other users using it without being able to minimize it to system tray or some other area in the bar is a deal-breaker.

**Comment by davidhealey** [comment]
Just to add to the feedback on this issue. Since the tray icon is no more I switched to a different torrenting app - I know it's ultimately a Gnome issue but the effect is you lost a user.

**Comment by guglovich** [comment]
If the transition from client to another client would be with the help of the program, it would be convenient. I have 4617 torrents on Trans. For some time I used GTK version from the community. Now I don't know.

**Reply by qsantos** [reply]
You can use  `transmission-qt`

**Reply by RuggedPineapple** [reply]
or just build it against gtk3. Its trivial to do. The only thing required is at the cmake configuration step instead of:
 cmake -B build -DCMAKE_BUILD_TYPE=Release

you do:
cmake -B build -DCMAKE_BUILD_TYPE=Release -DENABLE_GTK=ON -DUSE_GTK_VERSION=3
